icu4c/source/common/messagepattern.cpp

```diff
@@ -1164,7 +1164,7 @@ MessagePattern::setParseError(UParseError *parseError, int32_t index) {
     parseError->preContext[length]=0;
 
     // Set postContext to some of msg starting at index.
-    length=msg.length()-index;
+    length=msg.length()>index ? msg.length()-index : 0;
```

Comment:
Is this something that could reasonably happen? If not, I suggest instead adding the length condition to the first if-statement in this function and just do nothing at all if `index` is out of range.